packages/shared/src/Renderer.ts

```diff
@@ -1,3 +1,16 @@
+// this adds the used types if they haven't been set by @types/react or preact
+// In the future this may be fixable by accepting a JSX generic to every type
+// or a `createRenderer` function that implies a JSX generic.
```

Comment:
We can also fix it (later) by no longer importing this package in Angular InstantSearch (this would require splitting InstantSearch.js into two packages).

```suggestion
// Prevents type errors when using without a JSX implementation
// (e.g., Angular InstantSearch via InstantSearch.js)
// In the future, this may be fixable by accepting a JSX generic to every type
// or a `createRenderer` function that implies a JSX generic.
```